app/models/metric/ci_mixin/capture.rb

```diff
@@ -8,8 +8,10 @@ def perf_capture_object
   def queue_name_for_metrics_collection
     ems = if self.kind_of?(ExtManagementSystem)
             self
-          elsif self.respond_to?(:ext_management_system)
+          elsif respond_to?(:ext_management_system) && ext_management_system.present?
             ext_management_system
+          elsif respond_to?(:old_ext_management_system) && old_ext_management_system.present?
+            old_ext_management_system
```

Comment:
i dont think archive those(@simon3z right?) so from what i understand `old_ext_management_system` would either be `nil` and get caught by the exception below or be a live `ems` and fetch metrics correctly